### ISSUE TITLE
Multiple mapping with anyBody depends on orders declaration

### ISSUE THREAD

**Issue by silk-bahamut** [SpectoLabs/hoverfly-java#298]
## Description of the bug

The order of declaration of simulation depends on order even when using anyBody() method

## Steps to reproduce the issue

Here are two tests 
```

  @Test
  void shouldNotFail(Hoverfly hoverfly) {
    hoverfly.simulate(SimulationSource.dsl(
        HoverflyDsl.service("https://test.com")
            .post("/sameUrl")
            .body(HoverflyMatchers.matchesPartialJson(HttpBodyConverter.jsonWithSingleQuotes("{'attr':'value'}")))
            .willReturn(ResponseCreators.success())
            .post("/sameUrl")
            .anyBody()
            .willReturn(ResponseCreators.badRequest())
    ));
    Assertions.assertThat(new RestTemplate().exchange("https://test.com/sameUrl", HttpMethod.POST, new HttpEntity<>("""
            {
              "test": 1,
              "attr": "value",
              "list": [ "A", "B" ]
            }
            """), String.class))
        .isNotNull();
  }

  @Test
  void orderShouldNotMatter(Hoverfly hoverfly) {
    hoverfly.simulate(SimulationSource.dsl(
        HoverflyDsl.service("https://test.com")
            .post("/sameUrl")
            .anyBody()
            .willReturn(ResponseCreators.badRequest())
            .post("/sameUrl")
            .body(HoverflyMatchers.matchesPartialJson(HttpBodyConverter.jsonWithSingleQuotes("{'attr':'value'}")))
            .willReturn(ResponseCreators.success())
    ));
    Assertions.assertThat(new RestTemplate().exchange("https://test.com/sameUrl", HttpMethod.POST, new HttpEntity<>("""
            {
              "test": 1,
              "attr": "value",
              "list": [ "A", "B" ]
            }
            """), String.class))
        .isNotNull();
  }

```

### Observed result

The order is critical to the good execution of tests when it should not.

### Expected result

Both test should work the same way.

## Additional relevant information
If not indicated above:
1. Hoverfly Java version: 0.16.1

**Comment by silk-bahamut** [comment]
It was my though too
In case of similar score, the order plays a role and the last one wins it all?

**Comment by tommysitu** [comment]
> It was my though too
> 
> In case of similar score, the order plays a role and the last one wins it all?

That's right.

**Comment by tommysitu** [comment]
I've raised a PR to fix this, thanks for your report!

**Comment by tommysitu** [comment]
fixed in [v0.17.0](https://github.com/SpectoLabs/hoverfly-java/releases/tag/0.17.0)
